### PR TITLE
Remove configurable registration step

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: spec/controllers/spree/checkout_controller_spec.rb [hunk 1]
```diff
@@ -8,47 +8,19 @@
...
-      context 'when authenticated as registered user' do
-        before { allow(controller).to receive(:spree_current_user) { user } }
-
-        it 'proceeds to the first checkout step' do
-          spree_get :edit, { state: 'address' }
-          expect(response).to render_template :edit
-        end
-      end
+    context 'when authenticated as registered user' do
+      before { allow(controller).to receive(:spree_current_user) { user } }
```
Comment (dkubb) [inline spec/controllers/spree/checkout_controller_spec.rb]: You could use `receive_messages(spree_current_user: user)` here.

File: spec/controllers/spree/checkout_controller_spec.rb [hunk 2]
```diff
@@ -8,47 +8,19 @@
...
 
-      context 'when authenticated as guest' do
-        it 'redirects to registration step' do
-          spree_get :edit, { state: 'address' }
-          expect(response).to redirect_to spree.checkout_registration_path
-        end
+      it 'proceeds to the first checkout step' do
+        spree_get :edit, { state: 'address' }
```
Comment (dkubb) [inline spec/controllers/spree/checkout_controller_spec.rb]: The `{` and `}` are not necessary here.

File: spec/controllers/spree/checkout_controller_spec.rb [hunk 3]
```diff
@@ -8,47 +8,19 @@
...
+        expect(response).to render_template :edit
```
Comment (dkubb) [inline spec/controllers/spree/checkout_controller_spec.rb]: `render_template` should be parenthesized.

File: spec/controllers/spree/checkout_controller_spec.rb (hunk 2 above)
Comment (dkubb) [inline spec/controllers/spree/checkout_controller_spec.rb]: Should spree_get be parenthesized too?

File: spec/controllers/spree/checkout_controller_spec.rb [hunk 4]
```diff
@@ -8,47 +8,19 @@
...
-      end
-
-      context 'when authenticated as guest' do
-        it 'proceeds to the first checkout step' do
-          spree_get :edit, { state: 'address' }
-          expect(response).to render_template :edit
-        end
+    context 'when authenticated as guest' do
+      it 'redirects to registration step' do
+        spree_get :edit, { state: 'address' }
```
Comment (dkubb) [inline spec/controllers/spree/checkout_controller_spec.rb]: Instance of https://github.com/MountainRoseHerbs/spree_auth_devise/pull/21/files#r85435778

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

**Pull request closed without merging**